### DISCUSSION TITLE
Deep Nested field Array -  Previous nested values dont get cleaned when inserting new item

### DISCUSSION THREAD

**Discussion by Bair95** [react-hook-form/react-hook-form#6028] [Q&A]
**Describe the bug**
First of all many thanks to all the contributor of this awesome project. 
You saved me so much time with this library that i lost the count.

I'm not sure if it's a regression or a new bug as i've seen some closed issues referring to this behavior.
I think it's very close to : https://github.com/react-hook-form/react-hook-form/issues/5696 
Feel free to discard this issue if it's the same.

When trying to insert an item in a deep nested field, it seems that the previous nested values aren't cleaned so the newly added option get filled with the previous value

**To Reproduce**

1. Go to https://codesandbox.io/s/angry-fog-01z49
2. Wait for the data to be populated (timeout=3s)
3. In the Section 0, Click on "Add New options list"
3. On the newly created field (at the top), click on "Add new suboption"
4. See that the nested value of the first field is "test" ( it should be "" )

**Codesandbox link (Required)**
https://codesandbox.io/s/angry-fog-01z49

**Expected Behavior**
Newly created value should be empty

**Comment by bluebill1049** [comment]
<img width="486" alt="Screen Shot 2021-07-21 at 6 52 24 pm" src="https://user-images.githubusercontent.com/10513364/126460659-81f39570-e236-4e55-858e-0723b6211fc7.png">

Thanks for the issue report, but seriously this does make my life any easier to reproduce an issue, can you simply the codesandbot to isolate the problem.

**Comment by Bair95** [comment]
Thanks for the quick response.
I'm sorry for this mess i've cleaned up the CSB before submitting but it didnt saved my changes.
I've updated it now it should be cleaner.
![image](https://user-images.githubusercontent.com/1491762/126462914-2f7baa71-8f71-4a25-a27b-2018af29b35d.png)

**Comment by bluebill1049** [comment]
I will try to get an example.

**Comment by bluebill1049** [comment]
here is a triple field array example: https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-rbcgt?file=/src/tripleNest.js see if this helps yours.

**Reply by bluebill1049** [reply]
did you see my example above? it's working for me, see what's missing in your example?

**Reply by Bair95** [reply]
I did review it carefully and the data structure in your example doesnt match the use case i described , even if when submitted the data structure is ok ( triple nesting in data object when submitting ) .
I duplicated your example with the data structure i have and still reproduce the behavior i described above, I honestly dont know why its happening.
( Append Nested -> Append Triple : the new value contains "bill" instead of the provided { label: { en : "" }} supplied to the append method.
https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-f4tn2

**Reply by bluebill1049** [reply]
I am taking a look at it.

**Reply by bluebill1049** [reply]
I found the issue, you are using Controller as the flat input value. Which was the root cause of the issue, I will patch this problem to unblock such usage.

**Reply by bluebill1049** [reply]
use `register` should get around your issue for now.

**Reply by Bair95** [reply]
Thank you again for spending time on this.
I can confirm that using manually registered input instead of Controlled one solves the issue.

**Reply by bluebill1049** [reply]
no worries. here is another walk around: https://codesandbox.io/s/hardcore-austin-xyfcz?file=/src/App.js

**Reply by bluebill1049** [reply]
also, this PR will fix the problem for you: https://github.com/react-hook-form/react-hook-form/pull/5958 should be release in the next week or two.

https://codesandbox.io/s/affectionate-cloud-s2pqq?file=/src/App.js

**Reply by bluebill1049** [reply]
thanks for your patient 🙏 I wish all users are nice and patient like yourself ❤️

**Reply by Bair95** [reply]
Good news for the PR, can't wait to see it merged :) 
Being patient is the least one can do when dealing with open-source free-of-charge well-maintained-library imho ^^
Keep up the good work :+1:

**Comment by bluebill1049** [comment]
This could be related: https://github.com/react-hook-form/react-hook-form/issues/6040 i will looking into a bit more today

**Reply by bluebill1049** [reply]
it was not a bug and confirmed.

**Comment by Bair95** [comment]
Many thanks for taking some of your time to investigate this.
From what i can see on #6040 the appended object is in fact empty. In this case, changing append({}) with append({name:""}) solves the problem: the new appended field has no previous value in it.

As far as I can tell, this is not the case in my example, the appended object isn't empty: 
`const addChoice = () => {
    let newOption = {
      label: { fr: "" },
      value: nestedListOptions.fields.length
    };
    nestedListOptions.append(newOption);
  };`

I tried debugging  this use case step by step and it seems that the input value is populated after the append but I still dont know why. 
(first "append" call contains the proper object with empty property).

**Comment by Raphaww** [comment]
Hello, im the autor of #6040 I've updated my example but from the previous responses I think its the same issue: using Controller with a nested flat objet like this:
`
  <Controller
      name={${name}.children.${index}.options.position_b}
      ...
  />
`
unfortunately I don't know how to change some of my fields not to use `<Controller/>` 

Thank your for your quick responses :)

**Reply by bluebill1049** [reply]
I have a fix already, i will test it out tonight with your use case.
